src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1178,12 +1178,11 @@ oop ShenandoahHeap::evacuate_object(oop p, Thread* thread) {
 
   // Try to install the new forwarding pointer.
   oop copy_val = cast_to_oop(copy);
-  ContinuationGCSupport::relativize_stack_chunk(copy_val);
-
   oop result = ShenandoahForwarding::try_update_forwardee(p, copy_val);
   if (result == copy_val) {
     // Successfully evacuated. Our copy is now the public one!
     shenandoah_assert_correct(nullptr, copy_val);
+    ContinuationGCSupport::relativize_stack_chunk(copy_val);
```

Comment:
Please move it before the assert, so that we are sure relativization did not foobar the `copy_val`. Asserts in these methods verify that we return good things.

Reply:
Okay, I had it that way originally, but then changed it to make sure we were relativizing a valid copy.